src/GeneratedHydrator/CodeGenerator/Visitor/HydratorMethodsVisitor.php

```diff
@@ -42,9 +42,9 @@ public function __construct(ReflectionClass $reflectedClass)
             $className = $property->getDeclaringClass()->getName();
 
             if ($property->isPrivate() || $property->isProtected()) {
-                $this->hiddenPropertyMap[$className][] = $property->getName();
+                $this->hiddenPropertyMap[$className][] = ObjectProperty::fromReflection($property);
```

Comment:
`$this->hiddenPropertyMap` should be documented as an `array<string, array<int, ObjectProperty>>`

Reply:
Right, I'll do it.